Source/Core/Core/DSP/DSPCore.h

```diff
@@ -338,35 +412,118 @@ struct DSPInitOptions
   DSPInitOptions() : capture_logger(new DefaultDSPCaptureLogger()) {}
 };
 
-// Initializes the DSP emulator using the provided options. Takes ownership of
-// all the pointers contained in the options structure.
-bool DSPCore_Init(const DSPInitOptions& opts);
-
-void DSPCore_Reset();
-void DSPCore_Shutdown();  // Frees all allocated memory.
-
-void DSPCore_CheckExternalInterrupt();
-void DSPCore_CheckExceptions();
-void DSPCore_SetExternalInterrupt(bool val);
-
-// sets a flag in the pending exception register.
-void DSPCore_SetException(ExceptionType exception);
-
 enum class State
 {
   Stopped,
   Running,
   Stepping,
 };
 
-int DSPCore_RunCycles(int cycles);
+class DSPCore
```

Comment:
I know a lot of our classes / structs don't do this but since this is a new class, should this follow the rule of 5?

Reply:
Done